lib/Date/Utility.pm

```diff
@@ -407,51 +401,34 @@ sub _build_is_a_weekday {
     return ($self->is_a_weekend) ? 0 : 1;
 }
 
-my $EPOCH_RE = qr/^-?[0-9]{1,13}$/;
-
 =head2 new
 
 Returns a Date::Utility object.
 
 =cut
 
-## no critic (ProhibitNewMethod)
-sub new {
-    my ($self, $params_ref) = @_;
-    my $new_params = {};
-
-    if (not defined $params_ref) {
-        $new_params->{epoch} = time;
-    } elsif (ref $params_ref eq 'Date::Utility') {
-        return $params_ref;
-    } elsif (ref $params_ref eq 'HASH') {
-        if (not($params_ref->{'datetime'} or $params_ref->{epoch})) {
-            confess 'Must pass either datetime or epoch to the Date object constructor';
-        } elsif ($params_ref->{'datetime'} and $params_ref->{epoch}) {
-            confess 'Must pass only one of datetime or epoch to the Date object constructor';
-        } elsif ($params_ref->{epoch}) {
-            #strip other potential parameters
-            $new_params->{epoch} = $params_ref->{epoch};
-
-        } else {
-            #strip other potential parameters
-            $new_params = _parse_datetime_param($params_ref->{'datetime'});
-        }
-    } elsif ($params_ref =~ $EPOCH_RE) {
-        $new_params->{epoch} = $params_ref;
-    } else {
-        $new_params = _parse_datetime_param($params_ref);
-    }
-
-    my $obj = $popular{$new_params->{epoch}};
+use constant EPOCH_RE         => qr/^-?[0-9]{1,13}$/o;
+use constant EPOCH_MAYBE_FRAC => qr/^-?[0-9]{1,13}(?:\.[0-9]+)?$/o;
```

Comment:
where is 13 digits limit comes from? don't care, just curious.

Reply:
So... it is apparently acceptable to produce a "date string" like `YYYYMMDDHHMMSS` which is 14 digits.  From comments in there,  I believe it also used to be acceptable to produce `YYYYMMDD` so it was clamped at `{9,13}`.